lib/simple_form/wrappers/root.rb

```diff
@@ -30,6 +30,7 @@ def html_classes(input, options)
         end
         css << (options[:wrapper_error_class] || @defaults[:error_class]) if input.has_errors?
         css << (options[:wrapper_hint_class] || @defaults[:hint_class]) if input.has_hint?
+        css << (options[:wrapper_valid_class] || @defaults[:valid_class]) if input.valid?
```

Comment:
Why we are using a new `valid?` method and not the `!has_errors`?

Reply:
Sorry no idea, this code is untouched and original by @aeberlin

Reply:
Only `!has_errors?` would add the `valid_class` to a new rendered form by default. Combine the two `!has_errors? && has_value?` adds the class only on form submit and if the value is present and valid.